src/Raven.Server/Documents/Patch/BlittableObjectInstance.cs

```diff
@@ -138,410 +144,589 @@ private bool TryGetValueFromDocument(BlittableObjectInstance parent, string prop
+                    : reader.GetFieldReaderFor(Encodings.Utf8.GetBytes(property));
+
+                switch (fieldReader.Type)
                 {
-                    // here we need to perform a manipulation in order to generate the object from the data
-                    if (fieldType.IsJson)
+                    case IndexEntryFieldType.Empty:
+                        value = string.Empty;
+                        break;
+                    case IndexEntryFieldType.Null:
+                        value = DynamicJsNull.ExplicitNull;
+                        break;
+                    case IndexEntryFieldType.TupleListWithNulls:
+                    case IndexEntryFieldType.TupleList:
                     {
-                        Lucene.Net.Documents.Field[] propertyFields = parent.LuceneDocument.GetFields(property);
-
-                        JsValue[] arrayItems =
-                            new JsValue[propertyFields.Length];
+                        if (fieldReader.TryReadMany(out var iterator) == false)
+                            goto case IndexEntryFieldType.Invalid;
 
-                        for (int i = 0; i < propertyFields.Length; i++)
+                        var tupleList = new List<string>(iterator.Count);
+                        while (iterator.ReadNext())
                         {
-                            var field = propertyFields[i];
-                            var stringValue = field.StringValue(parent.LuceneState);
+                            if (iterator.IsNull)
+                            {
+                                tupleList.Add(null);
+                            }
+                            else if (iterator.IsEmptyCollection)
+                            {
+                                throw new InvalidDataException("Tuple list cannot contain an empty string (otherwise, where did the numeric came from!)");
+                            }
+                            else
+                            {
+                                tupleList.Add(Encodings.Utf8.GetString(iterator.Sequence));
+                            }
+                        }
 
-                            var itemAsBlittable = parent.Blittable._context.Sync.ReadForMemory(stringValue, field.Name);
+                        value = FromObject(parent.Engine, tupleList);
+                    }
+                        break;
 
-                            arrayItems[i] = TranslateToJs(parent, field.Name, BlittableJsonToken.StartObject, itemAsBlittable);
-                        }
+                    case IndexEntryFieldType.Tuple:
+                    {
+                        if (fieldReader.TryReadTuple(out long lVal, out double dVal, out Span<byte> valueInEntry) == false)
+                            goto case IndexEntryFieldType.Invalid;
 
-                        value = FromObject(parent.Engine, arrayItems);
-                        return true;
+                        if (valueInEntry.Contains((byte)'.'))
```

Comment:
Do you need to check it this way? Cannot `TryReadTuple` return whether it read double or long?

Reply:
No, the tuple always has [string, double, long] value inside.